### PR TITLE
Add Speech Streaming usage documentation.

### PR DESCRIPTION
**Pull request by daspecster**
Speech streaming has some implications to it that I think could be good to talk about.

- What kind of inputs should we support or is just a file correct for now?
- Should a context manager be used outside of the client or in the client.stream_recognize() method itself?
- If `interum_results` is `True`, should we `yield` the results as we receive them?

### CODE REVIEW

**Comment by daspecster** [comment]
@tseaver @dhermes LMK if you have any feedback for this. I updated it with the changes from #2495.

File: docs/speech-usage.rst [hunk 1]
```diff
@@ -97,6 +97,7 @@ Great Britian.
      transcript: Hello, this is one test
      confidence: 0
 
+
```
Comment (dhermes) [inline docs/speech-usage.rst]: Why is this here?

File: docs/speech-usage.rst [hunk 2]
```diff
@@ -141,5 +142,69 @@ words to the vocabulary of the recognizer.
...
+option should be disabled.
+
+See: `Single Utterance`_
+
+.. code-block:: python
+
+    >>> sample = client.sample(source_uri='gs://my-bucket/recording.flac',
+    ...                        encoding=Encoding.FLAC,
+    ...                        sample_rate=44100)
+    >>> results = client.stream_recognize(sample, single_utterance=False)
```
Comment (dhermes) [inline docs/speech-usage.rst]: Why not

```python
sample.stream_recognize(single_utterance=False)
```

File: docs/speech-usage.rst [hunk 3]
```diff
@@ -141,5 +142,69 @@ words to the vocabulary of the recognizer.
...
+If you would like interim results to be returned as well as the final results,
+you can set the ``interim_results`` option to ``True``.
+
+  .. code-block:: python
+
+    >>> sample = client.sample(source_uri='gs://my-bucket/recording.flac',
+    ...                        encoding=Encoding.FLAC,
+    ...                        sample_rate=44100)
+    >>> results = client.stream_recognize(sample, interim_results=True)
+    >>> print results[0].alternatives.transcript
```
Comment (dhermes) [inline docs/speech-usage.rst]: Ditto here. A reader now has two unanswered questions: what type is `results[0]`? what type is `results[0].alternatives`?

File: docs/speech-usage.rst (hunk 1 above)
Comment (daspecster) [inline docs/speech-usage.rst]: Something weird happened when I rebased upstream master into it. Removed in the next push.

File: docs/speech-usage.rst [hunk 4]
```diff
@@ -83,5 +83,73 @@ data to text and returns alternative text transcriptons.
...
+.. note::
+    Streaming recognition requests are limited to 1 minute of audio.
+
+    See: https://cloud.google.com/speech/limits#content
+
+.. code-block:: python
+
+    >>> from google.cloud import speech
+    >>> client = speech.Client()
+    >>> results = client.stream_recognize('hello.flac', 'FLAC', 44100)
```
Comment (daspecster) [inline docs/speech-usage.rst]: I think I should make a class called `Alternatives`, which could then use the `Transcript` class. (locally I just have it returning an object ATM.)

File: docs/speech-usage.rst [hunk 5]
```diff
@@ -83,5 +83,73 @@ data to text and returns alternative text transcriptons.
...
+    >>> results[0].is_final
+    False
+    >>> results[2].is_final
+    True
+    >>> results[2].alternatives[0].transcript
+    hello
+    >>> results[2].alternatives[0].confidence
+    0.96976006031
```
Comment (daspecster) [inline docs/speech-usage.rst]: Yeah I could do that.  Should there be one example accessing the data from the type that's returned?

File: docs/speech-usage.rst [hunk 6]
```diff
@@ -83,5 +83,73 @@ data to text and returns alternative text transcriptons.
...
+
+For continuous speech containing more than one word, the ``single_utterance``
+option should be disabled.
```
Comment (daspecster) [inline docs/speech-usage.rst]: The default upstream is False. So it detects 'continuously'. I can reverse the logic for this, if you think that makes sense?

File: docs/speech-usage.rst (hunk 2 above)
Comment (daspecster) [inline docs/speech-usage.rst]: Yeah, I can do that. I should move that up on `async_recognize()` and `sync_recognize()` as well.

File: docs/speech-usage.rst (hunk 6 above)
Comment (dhermes) [inline docs/speech-usage.rst]: I don't mind the logic. Just make it clear in these docs what the default is. In both examples you've used an explicit value. Instead, use no value (i.e. the default) in one of the example and then be explicit with a non-default in the other example.

File: docs/speech-usage.rst (hunk 2 above)
Comment (dhermes) [inline docs/speech-usage.rst]: Right.

File: docs/speech-usage.rst [hunk 7]
```diff
@@ -83,5 +83,73 @@ data to text and returns alternative text transcriptons.
...
+        stability: 0.00999999977648
+    },
+    {
+        alternatives: {
+            transcript: 'hello'
+            confidence: 0.96976006031
+        }
+        is_final: true
+    }
+    ]
```
Comment (daspecster) [inline docs/speech-usage.rst]: Yeah so, the GAPIC call, with `interim_results=True` returns something like this...

```bash
[alternatives {
  transcript: "how"
}
stability: 0.00999999977648
, alternatives {
  transcript: "hello"
}
stability: 0.00999999977648
, alternatives {
  transcript: "hello"
  confidence: 0.96976006031
}
is_final: true
]
```

I took that out in a later version of this doc though. I'm not sure why you're seeing it?

Comment (dhermes) [inline docs/speech-usage.rst]: I had a REALLY OLD "Pending" comment that I didn't notice. Ignore

File: docs/speech-usage.rst (hunk 4 above)
Comment (dhermes) [inline docs/speech-usage.rst]: SGTM. So this doc precedes that implementation?

File: docs/speech-usage.rst (hunk 5 above)
Comment (daspecster) [inline docs/speech-usage.rst]: OK, sounds good.

File: docs/speech-usage.rst (hunk 2 above)
Comment (daspecster) [inline docs/speech-usage.rst]: SG, https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2516

**Comment by daspecster** [comment]
I'm going to close this and make a new PR with the whole shebang.
